From 5aebd1a8c122a30f950582832ab3d6cf1bc978f7 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Wed, 15 Jul 2020 23:38:25 +0100
Subject: [PATCH] Fitler out annoying QML warnings

Qt 5.15 introduces a new syntax for connections.

Framework code can't port away due to having to support Qt5.12 which
does not include the new syntax.

This filters out the category.

We should remove this once we depend on Qt5.15 in frameworks.

BUG: 424259
---
 shell/main.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/shell/main.cpp b/shell/main.cpp
index c5e6a358e..df822644a 100644
--- a/shell/main.cpp
+++ b/shell/main.cpp
@@ -27,6 +27,7 @@
 #include <QDBusConnection>
 #include <QDBusMessage>
 #include <QtQml/QQmlDebuggingEnabler>
+#include <QLoggingCategory>
 
 #include <KAboutData>
 #include <KQuickAddons/QtQuickSettings>
@@ -47,6 +48,21 @@
 #include <QDir>
 #include <QDBusConnectionInterface>
 
+static QLoggingCategory::CategoryFilter oldCategoryFilter;
+
+// Qt 5.15 introduces a new syntax for connections
+// framework code can't port away due to needing Qt5.12
+// this filters out the warnings
+// Remove this once we depend on Qt5.15 in frameworks
+void filterConnectionSyntaxWarning(QLoggingCategory *category)
+{
+    if (qstrcmp(category->categoryName(), "qt.qml.connections") == 0) {
+        category->setEnabled(QtWarningMsg, false);
+    } else if (oldCategoryFilter) {
+        oldCategoryFilter(category);
+    }
+}
+
 int main(int argc, char *argv[])
 {
     if (qEnvironmentVariableIsSet("PLASMA_ENABLE_QML_DEBUG")) {
@@ -70,6 +86,8 @@ int main(int argc, char *argv[])
 
     QQuickWindow::setDefaultAlphaBuffer(true);
 
+    oldCategoryFilter = QLoggingCategory::installFilter(filterConnectionSyntaxWarning);
+
     const bool qpaVariable = qEnvironmentVariableIsSet("QT_QPA_PLATFORM");
     KWorkSpace::detectPlatform(argc, argv);
     QApplication app(argc, argv);
-- 
GitLab

